test/fuzzy_object_store/fuzz_object.cpp

```diff
@@ -0,0 +1,569 @@
+/*************************************************************************
+ *
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include "fuzz_object.hpp"
+#include "fuzz_logger.hpp"
+#include "util.hpp"
+
+#include <realm/group.hpp>
+#include <realm/data_type.hpp>
+#include <realm/util/optional.hpp>
+#include <realm/object-store/shared_realm.hpp>
+#include <realm.hpp>
+#include <realm/index_string.hpp>
+
+#include <utility>
+
+using namespace realm;
+using namespace realm::util;
+
+// Max number of rows in a table. Overridden only by create_object() and only in the case where
+// max_rows is not exceeded *prior* to executing add_empty_row.
+const size_t max_rows = 100000;
+const size_t add_empty_row_max = REALM_MAX_BPNODE_SIZE * REALM_MAX_BPNODE_SIZE + 1000;
+
+unsigned char FuzzObject::get_next_token(State& s) const
+{
+    if (s.pos == s.str.size() || s.str.empty()) {
+        throw EndOfFile{};
+    }
+    return s.str[s.pos++];
+}
+
+void FuzzObject::create_table(Group& group, FuzzLog& log)
```

Comment:
we should cover primary keys at some point too

Reply:
Can we add it later? This is just the kick-off, we are not ending this project right now... in fact we could probably start collecting all these into issues and start implementing them.